src/lib/utils/convertPercentageColor.ts

```diff
@@ -0,0 +1,47 @@
+import { ColorValue } from 'react-native';
+
+export const RGB_PATTERN =
+  /^rgb\(\s*(100%|100|\d{1,2}(\.\d+)?%)\s*,\s*(100%|100|\d{1,2}(\.\d+)?%)\s*,\s*(100%|100|\d{1,2}(\.\d+)?%)\s*\)$/;
+export const RGBA_PATTERN =
+  /^rgba\(\s*(100%|100|\d{1,2}(\.\d+)?%)\s*,\s*(100%|100|\d{1,2}(\.\d+)?%)\s*,\s*(100%|100|\d{1,2}(\.\d+)?%)\s*,\s*(1|0(\.\d+)?|100%|100|\d{1,2}(\.\d+)?%)\s*\)$/;
+
+const percentTo255 = (percent: string) =>
+  Math.round(parseFloat(percent) * 2.55);
+
+const parseAlpha = (alpha: string) =>
+  alpha.endsWith('%') ? parseFloat(alpha) / 100 : parseFloat(alpha);
+
+function parsePercentageRGBColor(color: string): string | undefined {
+  const rgbMatch = RGB_PATTERN.exec(color);
+  const rgbaMatch = RGBA_PATTERN.exec(color);
```

Comment:
Perhaps we should test with a second regex only if the first one doesn't match. Regex operations can be quite costly, so avoiding unnecessary steps would be more efficient.

Reply:
@jakex7 I've merged these two patterns into one to reduce regex operations [1fbdbb1](https://github.com/software-mansion/react-native-svg/pull/2363/commits/1fbdbb183f627c7c963898044a8d4fdacbf76cc5)